tests/math/test_matrix_manipulation.py

```diff
@@ -1068,3 +1069,84 @@ def test_expand_vector_invalid_vector(self):
         """Test exception raised if incorrect sized vector provided."""
         with pytest.raises(ValueError, match="Vector parameter must be of length"):
             expand_vector(TestExpandVector.VECTOR1, [0, 1], 4)
+
+
+class TestSqrtMatrix:
+    """Tests for the sqrt_matrix function."""
+
+    # NOTE: make sure the matrix is positive definite
+    dm_list = [
+        np.array([[1, 0], [0, 1]]),
+        np.array([[1, 0], [0, 2]]),
+        np.array([[4, 2], [2, 3]]),
+    ]
+    shape_list = range(2, 10)
+    # NOTE: sqrt_matrix is frequently used by BlockEncode
+    #       here below are some test matrices that are used
+    #       in the BlockEncode tests
+    matrices = [
+        # 2x2 matrices
+        np.array([[0.1, 0.2], [0.3, 0.4]]),
+        # Non-square matrices
+        np.array([[0.1, 0.2, 0.3], [0.3, 0.4, 0.2]]),
+        # 3x3 matrix
+        np.array([[0.1, 0.2, 0.3], [0.3, 0.4, 0.2], [0.1, 0.2, 0.3]]),
+        # Identity-like matrices
+        np.array([[1, 0], [0, 1]]),
+        np.identity(3),
+        # Matrix with complex entries
+        0.2 * np.array([[0.3, 0.9539392j], [0.9539392j, -0.3]]),
+    ]
+
```

Comment:
It could be good to also add a test for a matrix that doesn't converge/doesn't have a sqrt defined.

Reply:
added quite some ill-behaving examples in our latest commits